cellpack/autopack/Analysis.py

```diff
@@ -927,6 +1010,335 @@ def PairCorrelationFunction_2D(self, x, y, S, rMax, dr):
 
         return (g_average, radii, interior_indices)
 
+    def run_analysis_workflow(
+        self,
+        input_path,
+        output_path,
+        ingr_key,
+        run_similarity_analysis=True,
+        get_parametrized_representation=True,
+        save_plots=False,
+        get_correlations=False,
+    ):
+        all_objs, all_pos_list = self.get_obj_dict(input_path)
+        self.ingr_key = ingr_key
+
+        if run_similarity_analysis:
+            self.run_similarity_analysis(
+                all_objs,
+            )
+
+        if get_parametrized_representation:
+            self.get_parametrized_representation(
+                all_pos_list=all_pos_list,
+                angular_spacing=numpy.pi / 64,
+                inner_mesh_path=self.inner_mesh_path,
+                outer_mesh_path=self.outer_mesh_path,
+                save_plots=save_plots,
+                get_correlations=get_correlations,
+            )
+
+    def get_obj_dict(self, input_path):
```

Comment:
This may be a personal preference, but I like it when functions are above the functions that call them. It doesn't functionally matter within a class, but I still find it easier to read code this way